tools/validators/instance_validator/validate/telemetry_validator.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validates telemetry messages against a building config file.
+
+The validator will call a specified callback function if at least one message
+is received for all of the entities in the building config file, or if the
+specfied timeout duration is reached."""
+
+import threading
+
+from validate import telemetry
+from validate import telemetry_error
+
+DEVICE_ID = "deviceId"
+TRANSLATION = "translation"
+STATES = "states"
+UNITS = "unit_values"
+
+class TelemetryValidator(object):
+  """Validates telemetry messages against a building config file.
+
+  Args:
+    entities: parsed entities from building config
+    timeout: validation timeout duration in seconds
+    callback: callback function to be called when the validation finishes
+  """
+
+  def __init__(self, entities, timeout, callback):
+    super().__init__()
+    self.entities = entities
+    self.timeout = timeout
+    self.callback = callback
+    self.validated_entities = {}
+    self.validation_errors = []
+
+  def StartTimer(self):
+    """Starts the validation timeout timer."""
+    threading.Timer(self.timeout, lambda: self.callback(self)).start()
+
+  def AllEntitiesValidated(self):
+    """Returns true if a message was received for every entity."""
+    return len(self.entities) == len(self.validated_entities)
+
+  def CallbackIfCompleted(self):
+    """Checks if all entities have been validated, and calls the callback."""
+    if self.AllEntitiesValidated():
+      self.callback(self)
+
+  def AddError(self, error):
+    """Adds a validation error."""
+    self.validation_errors.append(error)
+
+  def ValidateMessage(self, message):
+    """Validates a telemetry message."""
```

Comment:
add detail about what where the errors go.

Reply:
Done